addons/web/controllers/home.py

```diff
@@ -122,7 +122,18 @@ def web_login(self, redirect=None, **kw):
                 credential = {key: value for key, value in request.params.items() if key in CREDENTIAL_PARAMS and value}
                 credential.setdefault('type', 'password')
                 if request.env['res.users']._should_captcha_login(credential):
-                    request.env['ir.http']._verify_request_recaptcha_token('login')
+                    try:
+                        request.env['ir.http']._verify_request_recaptcha_token('login')
+                    except:
+                        # We don't want to tell the user whether the captcha blocked them or not.
+                        # This is to make it harder for bots to brute force users since they won't know
+                        # whether the password is incorrect or whether the captcha is blocking them.
+                        #
+                        # However this may be revised in the future as the captcha can not be implemented
+                        # on RPC login attempts. Therefore this will only protect against most bot attacks
+                        # but not targeted ones.
+                        _logger.info('Captcha failed for user %s IP: %s', credential.get('login', '?'), request.httprequest.remote_addr)
+                        raise odoo.exceptions.AccessDenied()
```

Comment:
Quit those long comments, if I wanna read *why* this code is here, I'll read the commit message. Also it kinda defeats the purpose of the `AccessDenied` if you publicly advertise (the code is still public) that RPC login is not subject to the problem.

Btw you can chain exceptions, the user will receive an empty "403 - Forbidden" but we still keep the traceback in the logs.
```suggestion
                    try:
                        request.env['ir.http']._verify_request_recaptcha_token('login')
                    except UserError as exc:
                        raise AccessDenied() from exc
```

Btw `_verify_request_recaptcha_token` could be the one raising AccessDenied in the first place.